vector/src/main/java/im/vector/app/features/share/IncomingShareFragment.kt

```diff
@@ -71,13 +72,28 @@ class IncomingShareFragment @Inject constructor(
         setupToolbar(incomingShareToolbar)
         attachmentsHelper = AttachmentsHelper(requireContext(), this).register()
 
+        viewModel.observeViewEvents {
+            when (it) {
+                is IncomingShareViewEvents.ShareToRoom            -> handleShareToRoom(it)
+                is IncomingShareViewEvents.EditMediaBeforeSending -> handleEditMediaBeforeSending(it)
+                is IncomingShareViewEvents.MultipleRoomsShareDone -> handleMultipleRoomsShareDone(it)
+            }.exhaustive
+        }
+
         val intent = vectorBaseActivity.intent
         val isShareManaged = when (intent?.action) {
-            Intent.ACTION_SEND -> {
+            Intent.ACTION_SEND          -> {
                 var isShareManaged = attachmentsHelper.handleShareIntent(requireContext(), intent)
                 if (!isShareManaged) {
                     isShareManaged = handleTextShare(intent)
                 }
+
+                // Direct share
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P && intent.hasExtra(Intent.EXTRA_SHORTCUT_ID)) {
```

Comment:
is it mandatory to check SDK_INT here? In old OSs, `intent.hasExtra(Intent.EXTRA_SHORTCUT_ID)` should always return `false`, no?

Reply:
It is like this in the [example](https://github.com/googlecodelabs/android-direct-share/blob/f5be36dad6676e4fcb7f2a8f7a285466c7055c6a/direct-share-done/app/src/main/java/com/example/android/directshare/SendMessageActivity.kt#L113). It shouldn't be necessary, however.